### ISSUE TITLE
bug: React Native useMutation hook passing in an undefined input to server procedure, the input is clearly defined on the component

### ISSUE THREAD

**Issue by buebos** [trpc/trpc#3882]
### Provide environment information

Server dir output:

System:
    OS: Windows 10 10.0.19045
    CPU: (12) x64 AMD Ryzen 5 3600 6-Core Processor
    Memory: 3.14 GB / 15.91 GB
  Binaries:
    Node: 16.14.0 - C:\Program Files\nodejs\node.EXE
    npm: 9.3.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (110.0.1587.57)
    Internet Explorer: 11.0.19041.1566
  npmPackages:
    @trpc/server: ^10.11.0 => 10.11.0
    typescript: ^4.9.3 => 4.9.4

React Native app dir output:

System:
    OS: Windows 10 10.0.19045
    CPU: (12) x64 AMD Ryzen 5 3600 6-Core Processor
    Memory: 3.06 GB / 15.91 GB
  Binaries:
    Node: 16.14.0 - C:\Program Files\nodejs\node.EXE
    npm: 9.3.0 - ~\Desktop\software-dev\la-linea-app\la-linea-mobile\node_modules\.bin\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (110.0.1587.57)
    Internet Explorer: 11.0.19041.1566
  npmPackages:
    @tanstack/react-query: ^4.24.4 => 4.24.4
    @trpc/client: ^10.11.0 => 10.11.0
    @trpc/react-query: ^10.11.0 => 10.11.0
    @trpc/server: ^10.11.0 => 10.11.0
    react: 18.1.0 => 18.1.0
    typescript: ^4.9.4 => 4.9.4

### Describe the bug

Hello! Currently I'm trying trpc and I liked it so much, I tried to use it in my React Native app with Express backend as the documentation shows when combining with React. Everything with the 'useQuery' hook work just fine and as expected, except for when I try to use mutation procedure calls with an input.
When calling the 'trpc.myRoute.useMutation()' hook and it's method 'mutate' or 'mutateAsync' (the both of them) the trpc client is sending an undefined value instead of the string value I clearly assigned and defined in the code, I tried some solutions but non of them work and I didn't find too much support online.
I also checked types and my tsconfig.json and everything is as it should be, with strict on true and strictNullCheck on true.  Actually, that shouldn't be an issue because all the types mark down as expected.
If trpc does not support react native please let me know I would feel like an idiot hahaha, thanks for reading and helping if you can. I used an incorrect reproduce link because to reproduce the project you would have to install kinda lot of libs on client and server project.
Here is my code:

Server:

```typescript
// /server/trpc
export const createContext = ({
    req
}: trpcExpress.CreateExpressContextOptions) => {
    const ctxSchema = z.object({
        authorization: z.string().optional(),
        locale: z.string().optional()
    });

    const incomingHeadersCtx = {
        authorization: req.headers.authorization,
        locale: req.headers.locale
    };

    console.log(req.originalUrl);
    console.log(req.url);
    console.log(req.body);

    const ctx = ctxSchema.parse(incomingHeadersCtx);

    return ctx;
};
const trpc = initTRPC.context<typeof createContext>().create({});
export const router = trpc.router;
export const middleware = trpc.middleware;
export const publicProcedure = trpc.procedure;

export const appRouter = router({
    auth: authRouter, // contains a router with authGoogle
    report: reportRouter,
    ping: publicProcedure.input(z.string()).query(() => 'pong')
});
export type AppRouter = typeof appRouter;

```

```typescript
// server/procedures/authGoogle
// public procedure just an object of trpc.procedure
export const authGoogle = publicProcedure
    .input((val) => {
        console.log(val);  // receiving undefined

        if (typeof val === 'string') return val;

        throw new TRPCError({
            code: 'BAD_REQUEST',
            message: 'Expected string'
        }); // throws this error to client everytime
    })
    .mutation(async ({ input, ctx }) => {
        const googleId = await validateGoogleId(input);
        if(!googleId) throw new TRPCError({code: 'BAD_REQUEST', message: 'Invalid token' })

        return {
            authorization: signAuthToken({
                role: UserRole.USER,
                user_id: userRegistered.id
            })
        };
    });
```

RN Client:

```typescript
// rnapp/src/api/trpc.ts
import { createTRPCReact } from '@trpc/react-query';
import type { AppRouter } from '../../../server/trpc';

export const trpc = createTRPCReact<AppRouter>();
```

```tsx
// rnapp/App.tsx
...
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { httpBatchLink } from '@trpc/client';
import { trpc } from './src/api/trpc';

const App = () => {
   const [queryClient] = useState(() => new QueryClient());
   const [trpcClient] = useState(() =>
      trpc.createClient({
         links: [
            httpBatchLink({
               url: 'http://10.0.2.2:1010/trpc', // this is an equivalent url to localhost for the android simulator I'm running my app on
               headers() {
                  return {
                     authorization: getUserCredentials()
                  };
               },
            }),
         ],
      })
   );

   useEffect(() => {
      return initNotifeeEvtHandler();
   }, []);

   return (
      /**
       * Handler for the gestures on screen. It acts like a normal View
       * component
       */
      <GestureHandlerRootView style={{ flex: 1 }}>
         <SafeAreaProvider>
            
            <trpc.Provider client={trpcClient} queryClient={queryClient}>
               <QueryClientProvider client={queryClient}>
                  {/* Context for states required in most sections of the app */}
                  <AppContextContainer>
                     <RequestsContextProvider>
                        <AuthNavigator />
                     </RequestsContextProvider>
                  </AppContextContainer>
               </QueryClientProvider>
            </trpc.Provider>
         </SafeAreaProvider>
      </GestureHandlerRootView>
   );
};

```

```tsx
// rnapp/src/sections/auth/GoogleSigninBtn.tsx
const GoogleSigninBtn = ({ text }: Props) => {
   const { theme } = useContext(AppContext);
   const googleAuth = trpc.auth.google.useMutation();
   const ping = trpc.ping.useQuery('ping'); // test route

   /**
    * Returns 'pong' as expected and the url it fetched contains the input param of 'ping': 
    *  /trpc/ping?batch=1&input=%7B%220%22%3A%22ping%22%7D
    */
   if (ping.data) console.log(ping.data);

   const signin = async () => {
      try {
         // Method from a react native lib
         const { idToken } = await GoogleSignin.signIn();

         if (!idToken) return;
        
        console.log(idToken) // Outputs a string token indeed

        // At this poing idToken it's indeed a string but when I send the backend request the url consulted is:
        // /trpc/auth.google?batch=1
        // note that it does not have an input param, I have also try to mutate with a fixed wrong string like: 'example'
        // but the server also responds with the input error saying it's undefined
         const res = await googleAuth.mutateAsync(idToken);

         setUserCredentials(res.authorization);

         console.log('Success');
      } catch (e) {
         // Always throws the error I setted in backend: [TRPCClientError: Expected string]
         console.log(e);
      }
   };


   return (
      <CustomTouchable
         style={[
            {
               backgroundColor: theme.bgColor[1],
               maxHeight: 40,
               borderRadius: 5,
               alignItems: 'center',
               justifyContent: 'center',
               flexDirection: 'row',
               width: '100%',
               height: 40,
               marginTop: 20,
            },
         ]}
         onPress={signin}
      >
         <Image
            source={GoogleLogo}
            style={{
               width: 25,
               height: 25,
               position: 'absolute',
               top: 7.5,
               left: 7.5,
            }}
            resizeMethod="resize"
         />
         <Text lightFont center shadow>
            {text}
         </Text>
      </CustomTouchable>
   );
};

export default GoogleSigninBtn;`
```

### Link to reproduction

https://codesandbox.io/s/smoosh-meadow-3muc6r?file=/src/App.tsx

### To reproduce

1. Setup React Native project
2. Install trpc and libs required for react on that project
3. Setup a backend with express and trpc
4. Create a route with mutation method on procedure
5. Try to use the route from a component in the react native app

### 👨‍👧‍👦 Contributing

- [ ] 🙋‍♂️ Yes, I'd be down to file a PR fixing this bug!

**Comment by juliusmarminge** [comment]
The sandbox is just a plain react native app without any trpc actions

**Comment by KATT** [comment]
Thanks for the report, this does indeed look strange, but I do doubt that this is a problem with tRPC.

tRPC does work with RN.

Would you be able to reproduce this using https://github.com/t3-oss/create-t3-turbo?

**Comment by buebos** [comment]
It was an express middleware I forgot to take off my code!😶😔 Sorry for wasting your time

Hi! Thanks for replying to my issue and sorry for taking so long to answer you. I made some tests by cloning the repo you gave me and everything worked just fine. I also made another react native project and failed on mutation reqs. So I thought to myself "fuck this is so weird, I also doubt that is trpc" so I assumed there was something wrong in the server and indeed. Took off this middleware for url encoding on express:

```typescript
app.use(express.urlencoded({ extended: false }));
```

And worked perfectly. I feel dumb HAH, again thanks for answering me and congrats to such an awesome project. I will try to keep learning to take out the best of it! Sorry for not attaching a valid sandbox too. I will close this issue
